tests/src/test/java/alluxio/master/JournalShutdownIntegrationTest.java

```diff
@@ -97,21 +97,22 @@ public void run() {
   private static final int TEST_NUM_MASTERS = 3;
   private static final long TEST_TIME_MS = Constants.SECOND_MS;
 
-  private ClientThread mCreateFileThread = null;
+  private ClientThread mCreateFileThread;
   /** Executor for running client threads. */
-  private final ExecutorService mExecutorsForClient = Executors.newFixedThreadPool(1);
-  private Configuration mMasterConfiguration = null;
+  private ExecutorService mExecutorsForClient;
+
+  public SystemPropertyRule mDisableHdfsCacheRule =
+      new SystemPropertyRule("fs.hdfs.impl.disable.cache", "true");
 
   @After
   public final void after() throws Exception {
     mExecutorsForClient.shutdown();
-    System.clearProperty("fs.hdfs.impl.disable.cache");
     ConfigurationTestUtils.resetConfiguration();
   }
 
   @Before
   public final void before() throws Exception {
-    System.setProperty("fs.hdfs.impl.disable.cache", "true");
+    mExecutorsForClient = Executors.newFixedThreadPool(1);
```

Comment:
is this change related to thread-leaking? what is the difference between creating a threadpool per test case vs. per class?